lib/assert.js

```diff
@@ -103,8 +104,21 @@ x.throws = function (fn, err, msg) {
 			}
 		}, err, msg);
 
+		if (!isError(result)) {
+			throw result;
+		}
```

Comment:
> assert.throws will validate that the function indeed throws an error (a string) and thus does not jump to line 113.

Which is correct. Used like that you're only asserting something was thrown. You don't care what it is.